Data-Engineer.md

```diff
@@ -0,0 +1,126 @@
+# Work Sample for Data Engineer
+
+To effectively solve the following data pipeline problems, it is essential to use a DAG (Directed Acyclic Graph) oriented tool. DAG tools like Pachyderm, Airflow, Dagster, etc., can help streamline data processing and management with tracking data lineage, ensuring data integrity, and minimizing errors during processing.
+
+To provide more context and clarity, including pipeline specs and diagrams can be helpful. These artifacts can help visualize the DAG and its components, provide information on how data flows through the pipeline, and highlight the dependencies between tasks.
+
+## Problem 1: Raw Data Processing
+
+**Objective**: Ingest and process raw stock market datasets.
+
+### Tasks:
+1. Download the ETF and stock datasets from the primary dataset available at https://www.kaggle.com/datasets/jacksoncrow/stock-market-dataset.
+2. Setup a data structure to retain all data from ETFs and stocks with the following columns.
+    ```
+    Symbol: string
+    Security Name: string
+    Date: string (YYYY-MM-DD)
+    Open: float
+    High: float
+    Low: float
+    Close: float
+    Adj Close: float
+    Volume: int
+    ```
+3. Convert the resulting dataset into a structured format (e.g. Parquet).
+
+## Problem 2: Feature Engineering
+
+**Objective**: Build some feature engineering on top of the dataset from Problem 1.
+
+### Tasks:
+1. Calculate the moving average of the trading volume (`Volume`) of 30 days per each stock and ETF, and retain it in a newly added column `vol_moving_avg`.
+2. Similarly, calculate the rolling median and retain it in a newly added column `adj_close_rolling_med`.
+3. Retain the resulting dataset into the same format as Problem 1, but in its own stage/directory distinct from the first.
```

Comment:
@woozyking, can we add bonus task here to ask the candidate to write unit tests for each of the functions they are building?